app/models/post.rb

```diff
@@ -2,4 +2,12 @@ class Post < ApplicationRecord
   belongs_to :user
   has_many :likes
   has_many :comments
+
+  def update_posts_counter
+    Post.find_by(:id, post_id).count.update
+  end
```

Comment:
[OPTIONAL]

- It would be nice to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_posts_counter` method every time a new post has been saved to the database, by this you can make the `update_posts_counter` method private which will increase the security.

```ruby
# app/models/post.rb
  after_save :update_posts_counter

  private

  def update_posts_counter
    user.increment!(:posts_counter)
  end
```

Reply:
Same question here. `user` or `User`?

Reply:
Hi @HENRYKC24,

- Same here. Rails recommend using `user` instead of using class name here.

Reply:
:+1: